connector_magento_product_by_sku/models/product.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Hibou Corp.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo.addons.component.core import Component
+
+
+class ProductImporter(Component):
+    _inherit = 'magento.product.product.importer'
+
+    def _create(self, data):
+        product = self._find_existing_product(data)
+        if product:
+            data['odoo_id'] = product.id
+        return super(ProductImporter, self)._create(data)
+
+    def _find_existing_product(self, data):
+        dc = data.get('default_code')
+        if dc:
+            product_model = self.env['product.product']
+            return product_model.search([('default_code', '=', dc)], limit=1)
+        return None
```

Comment:
Usually I prefer to use the mapper for this purpose so we don't need to touch the importer's flow.

Should be a mapping such as:

```python
    def _find_existing_product(self, record):
        dc = record.get('default_code')
        product_model = self.env['product.product']
        if dc:
            return product_model.search([('default_code', '=', dc)], limit=1)
        return product_model.browse()

    @only_create
    @mapping
    def odoo_id(self, record):
        """ Will bind the product on a existing one with the same SKU """
        product = self._find_existing_product(record)
        return {'odoo_id': product.id}
```

What do you think about it?

Reply:
I tried this, but an @mapping for odoo_id never actually runs.

Reply:
Do you still have the code with the mapping somewhere? Because it would be a bug in the Components API :scream: